src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java

```diff
@@ -380,14 +500,104 @@ private void sendEmailToCafcassCymru(CaseData caseData, String cafcassCymruEmail
                                          EmailTemplateNames template) {
         String dashboardUrl = manageCaseUrl + "/" + caseData.getId() + "#Case%20documents";
         emailService.send(
-                cafcassCymruEmailId,
-                template,
-                buildUploadDocuemntEmail(caseData, respondantName, dashboardUrl),
-                LanguagePreference.getPreferenceLanguage(caseData)
+            cafcassCymruEmailId,
+            template,
+            buildUploadDocuemntEmail(caseData, respondantName, dashboardUrl),
+            LanguagePreference.getPreferenceLanguage(caseData)
         );
 
     }
 
+    private UUID sendResponseDocumentViaPost(CaseData caseData,
+                                             Address address,
+                                             String name,
+                                             String authorisation,
+                                             List<Document> responseDocument) throws Exception {
+        List<Document> documents = new ArrayList<>();
+        //generate cover letter
+        List<Document> coverLetterDocs = serviceOfApplicationPostService.getCoverSheets(
+            caseData,
+            authorisation,
+            address,
+            name,
+            DOCUMENT_COVER_SHEET_SERVE_ORDER_HINT
+        );
+        if (CollectionUtils.isNotEmpty(coverLetterDocs)) {
+            documents.addAll(coverLetterDocs);
+        }
+
+        //cover should be the first doc in the list, append all order docs
+        documents.addAll(responseDocument);
+
+        return bulkPrintService.send(
+            String.valueOf(caseData.getId()),
+            authorisation,
+            LETTER_TYPE,
+            documents,
+            name
+        );
+    }
+
+    public List<Document> getCoverLetters(CaseData caseData, String auth, Address address, String name,
```

Comment:
Please move this piece of code to a generic method that can be used in SOA and Document management

Reply:
moved this method to service of application